drivers/dai/intel/dmic/dmic.c

```diff
@@ -850,9 +860,10 @@ static int dmic_pm_action(const struct device *dev, enum pm_device_action action
 	case PM_DEVICE_ACTION_RESUME:
 		dai_dmic_probe_wrapper(dev);
 		break;
-	case PM_DEVICE_ACTION_TURN_OFF:
 	case PM_DEVICE_ACTION_TURN_ON:
-		/* All device pm is handled during resume and suspend */
+		dai_dmic_irq_init();
+		break;
```

Comment:
You are not doing anything special here, just do the same thing you do for `PM_DEVICE_ACTION_TURN_OFF` and move that device initialization to the init funciton. Note that this state is only triggered when the power domain this device belongs  is resumed.